lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -296,6 +296,37 @@ def import_new_workflow_deprecated(self, trans, payload, **kwd):
         return self.__api_import_new_workflow( trans, payload, **kwd )
 
     @expose_api
+    def save( self, trans, id, payload, **kwd ):
+        """
+        POST /api/workflows/id/save
+        Save the workflow described by `workflow_data` with id `id`.
+        """
+        # Get the stored workflow
+        stored = self.__get_stored_workflow( trans, id )
+        try:
+            workflow, errors = self.workflow_contents_manager.update_workflow_from_dict(
+                trans,
+                stored,
+                payload,
+            )
+        except workflows.MissingToolsException as e:
+            return dict(
+                name=e.workflow.name,
+                message="This workflow includes missing or invalid tools. It cannot be saved until the following steps are removed or the missing tools are enabled.",
+                errors=e.errors,
+            )
```

Comment:
I don't think returning a `dict` here follows Galaxy tradition of raising an Exception if an API call fails, I'd rather remove the except block.